### PR TITLE
[ELY-489] Add JavaDoc for the 'org.wildfly.security.mechanism' package and sub packages

### PR DESCRIPTION
**Pull request by petrberan**
https://issues.redhat.com/browse/ELY-489

### CODE REVIEW

File: mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/DigestUtil.java [hunk 1]
```diff
@@ -232,10 +266,13 @@ public static byte[] userRealmPasswordDigest(MessageDigest messageDigest, String
     }
 
     /**
-     * Get array of password chars from TwoWayPassword
+     * Get array of password chars from TwoWayPassword.
      *
-     * @return
-     * @throws SaslException
```
Comment (fjuma) [inline mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/DigestUtil.java:238]: The `SaslException` import needs to be removed too. CI currently fails with this error:

```
/home/runner/work/wildfly-elytron/wildfly-elytron/mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/DigestUtil.java:33:8: Unused import - javax.security.sasl.SaslException. [UnusedImports]
```

Comment (petrberan) [reply mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/DigestUtil.java:238]: Sorry about that, that was a stupid oversight on my part. Fixed now

Comment (fjuma) [reply mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/DigestUtil.java:238]: No worries, @petrberan! Thanks for updating this!

**Comment by petrberan** [comment]
Hi @fjuma @Skyllarr can I ask for review please?

**Comment by petrberan** [comment]
@fjuma  Can I ask for a review please?

**Comment by fjuma** [comment]
@petrberan When you get a chance, please update this so it's based off the 2.x branch, thanks!

File: mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/PasswordDigestObtainer.java [hunk 2]
```diff
@@ -83,14 +97,30 @@ public PasswordDigestObtainer(CallbackHandler callbackHandler, String defaultUse
         this.skipRealmCallbacks = skipRealmCallbacks;
     }
 
+    /**
+     * the default username to use if a callback is not provided.
```
Comment (skyllarr) [inline mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/PasswordDigestObtainer.java]: @petrberan Just a total minor, the default username is returned only if other username was not obtained from callback, no? Like [here](https://github.com/wildfly-security/wildfly-elytron/blob/1.x/mechanism/digest/src/main/java/org/wildfly/security/mechanism/digest/PasswordDigestObtainer.java#L152). So "return the username obtained from callback or the default username". Same for **realm** parameter below

File: mechanism/oauth2/src/main/java/org/wildfly/security/mechanism/oauth2/OAuth2Server.java [hunk 3]
```diff
@@ -52,12 +52,26 @@ public class OAuth2Server {
     private final Map<String, ?> serverConfig;
     private ElytronMessages log;
 
+    /**
+     * Constructs a new {@code OAuth2Server} instance.
+     *
+     * @param callbackHandler the callback handler for obtaining the Bearer token.
```
Comment (skyllarr) [inline mechanism/oauth2/src/main/java/org/wildfly/security/mechanism/oauth2/OAuth2Server.java]: @petrberan  Just a minor, the token is obtained from the initialClientMessage of the client AFAIK. The callbackHandler will verify the token

File: mechanism/scram/src/main/java/org/wildfly/security/mechanism/scram/ScramServer.java [hunk 4]
```diff
@@ -240,6 +262,16 @@ public ScramInitialServerResult evaluateInitialResponse(final ScramInitialClient
         return new ScramInitialServerResult(new ScramInitialServerMessage(clientMessage, serverNonce, salt, iterationCount, messageBytes), password);
     }
 
+    /**
+     * Parses the final client message for the {@link ScramFinalClientMessage} from parsed information.
+     * Also checks if the message have all necessary properties.
```
Comment (skyllarr) [inline mechanism/scram/src/main/java/org/wildfly/security/mechanism/scram/ScramServer.java]: Just total minors,

s/Parses the final client message for the {@link ScramFinalClientMessage} from parsed information./Parses the final client message and constructs the {@link ScramFinalClientMessage} from this parsed information.

s/message have/message has

**Review by skyllarr — COMMENTED** [review]
Hi @petrberan , I'm very sorry for the delay in review. This looks good to me, I just added very minor comments. Pls also update the base branch of the PR to be the 2.x branch instead of the 1.x branch. Thanks a lot!

**Comment by petrberan** [comment]
Again, sorry for the delay on this one @fjuma @Skyllarr . Managed to address those comments above and rebased this on top of 2.x, also changing the target branch

File: mechanism/oauth2/src/main/java/org/wildfly/security/mechanism/oauth2/OAuth2InitialClientMessage.java [hunk 5]
```diff
@@ -27,24 +29,51 @@ public class OAuth2InitialClientMessage {
...
+     * @return the authorization information in form of a String.
+     */
     public String getAuth() {
         return auth;
     }
 
+    /**
+     * Returns whether the client provides a Bearer token.
+     *
+     * @return {@code True} if the authorization information contains "Bearer", {@code false} otherwise.
```
Comment (fjuma) [inline mechanism/oauth2/src/main/java/org/wildfly/security/mechanism/oauth2/OAuth2InitialClientMessage.java:75]: s/True/true

**Review by fjuma — APPROVED** [review]
Looks great, @petrberan! Just added a small comment.

**Comment by fjuma** [comment]
@Skyllarr @ivassile Please review in the new year, thanks!

**Review by ivassile — APPROVED** [review]

**Pull request merged**